libdnf/dnf-context.c

```diff
@@ -1027,6 +1044,22 @@ dnf_context_set_cache_age(DnfContext *context, guint cache_age)
     priv->cache_age = cache_age;
 }
 
+
+/**
+ * dnf_context_set_cache_suffix_match:
+ * @context: a #DnfContext instance.
+ * @cache_suffix_match : %FALSE means the cache needs to be re-updated (default is false)
+ *
+ * Sets status of cache suffix
+ *
+ **/
+void
+dnf_context_set_cache_suffix_match(DnfContext *context, gboolean cache_suffix_match)
```

Comment:
ah.... The way you were saying it.....  made me realize that all of the checks I have done was to make sure the suffix matches, but did not realize that once the repo location is set, if the cache_dir was not present, it will be downloaded anyway.... I guess.... I learned the lesson the hard way T_T. All of the previous work seemed unnecessary.....

Thanks for the comment, will do the fix and it seems also make sense to discard many of the changes listed earlier.....